pkg/controller/servicebindingrequest/mapper.go

```diff
@@ -21,17 +120,61 @@ func (m *sbrRequestMapper) Map(obj handler.MapObject) []reconcile.Request {
 		"Object.Namespace", obj.Meta.GetNamespace(),
 		"Object.Name", obj.Meta.GetName(),
 	)
-	toReconcile := []reconcile.Request{}
 
-	sbrNamespacedName, err := getSBRNamespacedNameFromObject(obj.Object)
+	namespacedNamesToReconcile := make(namespacedNameSet)
+
+	if isServiceBindingRequest(obj.Object) {
+		requests := []reconcile.Request{
+			{NamespacedName: convertToNamespacedName(obj.Meta)},
+		}
+		log.Debug("current resource is a SBR", "Requests", requests)
+		return requests
+	}
+
+	// note(isutton): The client handles retries on the operator behalf, so only unrecoverable errors
+	// are left.
+	//
+	// please see https://github.com/isutton/service-binding-operator/blob/e17445570bd3889bcf7499142350a3b81463c6be/vendor/k8s.io/client-go/rest/request.go#L723-L812
+	sbrList, err := m.client.Resource(groupVersion).List(metav1.ListOptions{})
 	if err != nil {
-		log.Error(err, "on inspecting object for annotations for SBR object")
-		return toReconcile
+		log.Error(err, "listing SBRs")
+		return []reconcile.Request{}
 	}
-	if isNamespacedNameEmpty(sbrNamespacedName) {
-		log.Debug("not able to extract SBR namespaced-name")
-		return toReconcile
+
+ITEMS:
+	for _, item := range sbrList.Items {
+		namespacedName := convertToNamespacedName(&item)
+
+		sbr, err := convertToSBR(item.Object)
+		if err != nil {
+			log.Error(err, "converting unstructured to SBR")
+			continue ITEMS
+		}
+
+		if isSecret(obj.Object) && isSecretOwnedBySBR(obj.Meta, sbr) {
```

Comment:
Just asking, is there any logical reason why secret check is before Service and Application check?

Reply:
That's because we also need to check whether the secret is being "owned" by the SBR through its `.status.secret` field, which is not part of service and application checks.

Reply:
I believe this block could also continue to the next `ITEMS` element at that point, since it is not important where the relationship has been identified but only that it has been identified.

Reply:
Okay Igor!